src/provider.ts

```diff
@@ -88,8 +88,8 @@ export interface MarinaProvider {
   importTemplate(template: Template, changeTemplate?: Template): Promise<void>;
 
   // get next (change) address for the current selected account
-  getNextAddress(): Promise<AddressInterface>;
-  getNextChangeAddress(): Promise<AddressInterface>;
+  getNextAddress(constructorParams?: any): Promise<AddressInterface>;
```

Comment:
I would type it as `(string|number)[]` at least

Reply:
I supposed the params needed to be labeled and therefore I thought they would have been provided as an object.
If this is not actually needed, `(string|number)[]` makes totally sense to me.

Reply:
>  the params needed to be labeled

I don't get whay you mean by labeled. the `window.postMessage` API sends data as string and we de/serialize the messages as JSON, so we can't send stuff such as Buffer or other types, so we limited to number,string and maybe boolean, but Ionio constructor arguments are either JS string, number or Buffer.

From https://github.com/ionio-lang/ionio/pull/15 I made sure that we can pass bytes-like type hex encoded prefixed with `0x` to allow this

https://github.com/ionio-lang/ionio/pull/15